Tests/TensorFlowTests/OperatorTests/BasicTests.swift

```diff
@@ -707,6 +719,7 @@ final class BasicOperatorTests: XCTestCase {
         ("testConcatenation", testConcatenation),
         ("testVJPConcatenation", testVJPConcatenation),
```

Comment:
Minor, unrelated to this patch: I wonder why `testVJPConcatenation` isn't defined in `TensorAutoDiffTests.swift`. It should be!

Reply:
Can we talk about the structure of where gradient tests should be placed at? In the case of matrix and linear algebra operators they are in there own test files. Same for layer tests, is `autodifftests`, supposed to be the place for all other gradient tests?

Reply:
Interesting point. I don't care strongly about how derivative test organization right now.

I do think that placing derivative tests right after "original function" tests is a bit nicer for locality. This means we could fully remove `TensorAutoDiffTests.swift`.

Maybe others feel differently though.